### PR TITLE
AzureStorage: CancellationTokens + Core Alpha 940

### PR DESCRIPTION
**Pull request by DavidBoike**

### CODE REVIEW

**Comment by DavidBoike** [comment]
Updated this today, and cherry-picked in the GitHubSync PR. Currently a few test failures and I'm not sure why.

**Comment by DavidBoike** [comment]
Actually got the Outbox stuff working. Ready to review.

**Review by adamralph — APPROVED** [review]

**Review by andreasohlund — APPROVED** [review]

File: src/NServiceBus.Persistence.AzureTable/Outbox/OutboxRecord.cs [hunk 1]
```diff
@@ -44,5 +46,13 @@ public void SetAsDispatched()
             Operations = Array.Empty<TransportOperation>();
             DispatchedAt = DateTimeOffsetHelper.ToWireFormattedString(DateTimeOffset.UtcNow);
         }
+
+        class StorageTransportOperation
```
Comment (mauroservienti) [inline src/NServiceBus.Persistence.AzureTable/Outbox/OutboxRecord.cs:50]: Does moving from `TransportOperation` to `StorageTransportOperation` impact wire compatibility? e.g. let's imagine that an endpoint is upgraded to V8 and at startup look for not yet dispatched messages. Would that work?

Comment (DavidBoike) [reply src/NServiceBus.Persistence.AzureTable/Outbox/OutboxRecord.cs:50]: Yep, it should. StorageTransportOperation is basically the V7 representation of those classes. After the change from a Dictionary<string, string> to DispatchProperties, JSON.NET would serialize DispatchProperties correctly as a dictionary, but would not deserialize the "dictionary" back to DispatchProperties. JSON.NET does not do any of the `$type` properties unless you have a situation like List<BaseClass> that is full of instances of more specialized classes that inherit from BaseClass. So on the wire, it's all the same.

**Review by mauroservienti — APPROVED** [review]

**Pull request merged**